Tools/clinic/clinic.py

```diff
@@ -4288,7 +4288,7 @@ def eval_ast_expr(
         globals: dict[str, Any],
         *,
         filename: str = '-'
-) -> FunctionType:
+) -> Any:
```

Comment:
Should we add a comment as to why `Any` is the correct annotation here?

Reply:
You weren't going to make finally closing that issue easy, were you? ;)

Let me know if a5103be is too verbose...!

Reply:
Hmm, actually, now that I look at it again, the comment feels like it duplicates the docstring a little bit... not sure if it's worth it? I'll leave it up to you!

Reply:
Sorry, you're right. The docstring should be enough!

Reply:
Okay -- removed the comment again and tweaked the docstring slightly. How's it look now?